compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -1904,14 +1904,19 @@ public void visit(BLangAnnotAccessExpr annotAccessExpr) {
 
     @Override
     public void visit(BLangStatementExpression bLangStatementExpression) {
-        if (bLangStatementExpression.stmt.getKind() == NodeKind.BLOCK) {
-            BLangBlockStmt bLangBlockStmt = (BLangBlockStmt) bLangStatementExpression.stmt;
-            for (int i = 0; i < bLangBlockStmt.stmts.size(); i++) {
-                BLangStatement stmt = bLangBlockStmt.stmts.remove(i);
-                bLangBlockStmt.stmts.add(i, rewrite(stmt, env));
+        BLangStatement exprStmt = bLangStatementExpression.stmt;
+        if (exprStmt.getKind() == NodeKind.BLOCK) {
+            BLangBlockStmt bLangBlockStmt = (BLangBlockStmt) exprStmt;
+            if (bLangBlockStmt.isLetExpr) {
+                bLangStatementExpression.stmt = rewrite(bLangBlockStmt, env);
+            } else {
+                for (int i = 0; i < bLangBlockStmt.stmts.size(); i++) {
+                    BLangStatement stmt = bLangBlockStmt.stmts.remove(i);
+                    bLangBlockStmt.stmts.add(i, rewrite(stmt, env));
+                }
             }
```

Comment:
HBT this? we can avoid the nested if-else.
```suggestion
        if (exprStmt.getKind() == NodeKind.BLOCK && !((BLangBlockStmt) exprStmt).isLetExpr) {
            BLangBlockStmt bLangBlockStmt = (BLangBlockStmt) exprStmt;
            for (int i = 0; i < bLangBlockStmt.stmts.size(); i++) {
                BLangStatement stmt = bLangBlockStmt.stmts.remove(i);
                bLangBlockStmt.stmts.add(i, rewrite(stmt, env));
            }
```